mempool/mempool.go

```diff
@@ -31,6 +31,12 @@ type PreCheckFunc func(types.Tx) error
 // transaction doesn't require more gas than available for the block.
 type PostCheckFunc func(types.Tx, *abci.ResponseCheckTx) error
 
+// TxInfo are parameters that get passed when attempting to add a tx to the
+// mempool.
+type TxInfo struct {
+	PeerID uint16
```

Comment:
Would it make sense to use the type `p2p.ID`? If not should we add a comment why this is necessary / better?